### ISSUE TITLE
Add support for cache instance family cache.m7g and cache-r7g

### ISSUE THREAD

**Issue by dvxam** [terraform-linters/tflint-ruleset-aws#543]
New node type has been released this February. https://aws.amazon.com/about-aws/whats-new/2023/02/amazon-ec2-m7g-r7g-instances/?nc1=h_ls

They are now supported by ElastiCache, and should be added the the  `validElastiCacheNodeTypes` list.

Here is up-to-date list:

```go
// rules/utils.go 
package rules

var validElastiCacheNodeTypes = map[string]bool{
	// https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/CacheNodes.SupportedTypes.html
	"cache.t2.micro":      true,
	"cache.t2.small":      true,
	"cache.t2.medium":     true,
	"cache.t3.micro":      true,
	"cache.t3.small":      true,
	"cache.t3.medium":     true,
	"cache.t4g.micro":     true,
	"cache.t4g.small":     true,
	"cache.t4g.medium":    true,
	"cache.m3.medium":     true,
	"cache.m3.large":      true,
	"cache.m3.xlarge":     true,
	"cache.m3.2xlarge":    true,
	"cache.m4.large":      true,
	"cache.m4.xlarge":     true,
	"cache.m4.2xlarge":    true,
	"cache.m4.4xlarge":    true,
	"cache.m4.10xlarge":   true,
	"cache.m5.large":      true,
	"cache.m5.xlarge":     true,
	"cache.m5.2xlarge":    true,
	"cache.m5.4xlarge":    true,
	"cache.m5.12xlarge":   true,
	"cache.m5.24xlarge":   true,
	"cache.m6g.large":     true,
	"cache.m6g.xlarge":    true,
	"cache.m6g.2xlarge":   true,
	"cache.m6g.4xlarge":   true,
	"cache.m6g.8xlarge":   true,
	"cache.m6g.12xlarge":  true,
	"cache.m6g.16xlarge":  true,
	"cache.m7g.large":     true,
	"cache.m7g.xlarge":    true,
	"cache.m7g.2xlarge":   true,
	"cache.m7g.4xlarge":   true,
	"cache.m7g.8xlarge":   true,
	"cache.m7g.12xlarge":  true,
	"cache.m7g.16xlarge":  true,
	"cache.r3.large":      true,
	"cache.r3.xlarge":     true,
	"cache.r3.2xlarge":    true,
	"cache.r3.4xlarge":    true,
	"cache.r3.8xlarge":    true,
	"cache.r4.large":      true,
	"cache.r4.xlarge":     true,
	"cache.r4.2xlarge":    true,
	"cache.r4.4xlarge":    true,
	"cache.r4.8xlarge":    true,
	"cache.r4.16xlarge":   true,
	"cache.r5.large":      true,
	"cache.r5.xlarge":     true,
	"cache.r5.2xlarge":    true,
	"cache.r5.4xlarge":    true,
	"cache.r5.12xlarge":   true,
	"cache.r5.24xlarge":   true,
	"cache.r6g.large":     true,
	"cache.r6g.xlarge":    true,
	"cache.r6g.2xlarge":   true,
	"cache.r6g.4xlarge":   true,
	"cache.r6g.8xlarge":   true,
	"cache.r6g.12xlarge":  true,
	"cache.r6g.16xlarge":  true,
	"cache.r6gd.xlarge":   true,
	"cache.r6gd.2xlarge":  true,
	"cache.r6gd.4xlarge":  true,
	"cache.r6gd.8xlarge":  true,
	"cache.r6gd.12xlarge": true,
	"cache.r6gd.16xlarge": true,
	"cache.r7g.large":     true,
	"cache.r7g.xlarge":    true,
	"cache.r7g.2xlarge":   true,
	"cache.r7g.4xlarge":   true,
	"cache.r7g.8xlarge":   true,
	"cache.r7g.12xlarge":  true,
	"cache.r7g.16xlarge":  true,
	"cache.m1.small":      true,
	"cache.m1.medium":     true,
	"cache.m1.large":      true,
	"cache.m1.xlarge":     true,
	"cache.m2.xlarge":     true,
	"cache.m2.2xlarge":    true,
	"cache.m2.4xlarge":    true,
	"cache.c1.xlarge":     true,
	"cache.t1.micro":      true,
}

var previousElastiCacheNodeTypes = map[string]bool{
	// https://aws.amazon.com/elasticache/previous-generation/?nc1=h_ls
	"c1": true,
	"m1": true,
	"m2": true,
	"m3": true,
	"r3": true,
	"t1": true,
}
```

New lines are following ones:
```
	"cache.m7g.xlarge":    true,
	"cache.m7g.2xlarge":   true,
	"cache.m7g.4xlarge":   true,
	"cache.m7g.8xlarge":   true,
	"cache.m7g.12xlarge":  true,
	"cache.m7g.16xlarge":  true,
        .... 
	"cache.r7g.large":     true,
	"cache.r7g.xlarge":    true,
	"cache.r7g.2xlarge":   true,
	"cache.r7g.4xlarge":   true,
	"cache.r7g.8xlarge":   true,
	"cache.r7g.12xlarge":  true,
	"cache.r7g.16xlarge":  true,
```

I can do the Merge Request if added a contributor.

**Comment by bendrucker** [comment]
Yes, please send a PR

**Comment by dvxam** [comment]
@bendrucker I would need to be added as contributor. PR is ready on my side, but I can't push to this repo.

**Comment by bendrucker** [comment]
You can send PRs from forks

**Comment by dvxam** [comment]
Here is the associated PR https://github.com/terraform-linters/tflint-ruleset-aws/pull/545
